app/uk/gov/hmrc/transitmovementsrouter/controllers/MessagesController.scala

```diff
@@ -103,7 +100,14 @@ class MessagesController @Inject() (
     implicit request =>
       implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromRequest(request)
       (for {
-        _ <- parseAndLogUpscanResponse(request.body)
+        upscanResponse <- parseAndLogUpscanResponse(request.body)
+        objectStoreResponse <- mapToOptionalResponse[
+          ObjectStoreError,
+          ObjectSummaryWithMd5
+        ](
+          objectStoreService.addMessage(upscanResponse.downloadUrl.get, movementId, messageId)
+        )
```

Comment:
Doing this completely defeats the point of `EitherT` -- because we've effectively converted a error value (left) to a success value (right). This line should just be

```scala
objectStoreResponse <- objectStoreService.addMessage(upscanResponse.downloadUrl.get, movementId, messageId).asPresentation
```

`asPresentation` does the conversion you defined in `ConvertError`, and will then short circuit the rest of the comprehension.

Reply:
Done